### ISSUE TITLE
HCAEnc.exe

### ISSUE THREAD

**Issue by Lakoyo** [OpenCGSS/DereTore#54]
How to use hcaenc.exe? I want to encrypt custom key to .hca file. When I drag and drop it to hcaenc.exe nothing happens. What's wrong?

**Comment by hozuki** [comment]
Since the questions asked are very similar, you can take a look at #51.

**Comment by Lakoyo** [comment]
Yes but still I don't know how to write command line correctly works for it. Please help me for typing :)

**Comment by Lakoyo** [comment]
Help please

**Comment by MiniatureHusk** [comment]
@Lakoyo:
in my case, it worked by drag and dropping HCACC.exe to the CMD window. I made sure to leave a space after ".exe" though. And then drag and drop your HCA file, this should specify exactly to where the input HCA file is. And if you want to save it next to your input HCA, then copy-paste the path of your input HCA and change the HCA's file name. So it should look like this

`C:\Users\CURRENT USER>D:\More\DereTore-master\Apps\Hcacc\bin\Release\hcacc.exe "D:\more\custom_song\input hca.hca" "D:\more\custom_song\output hca.hca" -ot 0` and just hit enter.

And actually, I want to point out that @hozuki 's answer on #51 raised another question for me. Which 16-hex halves go to where? I got a bit confused when they considered the second half of the key to be 00000000. Now, my problem would be,
what if the custom key's hex value was longer?
I'll make my own custom key for this example:
91280410300 = 1540BC86BC
so is key no. 1 `00000015`? or `40BC86BC`?
or is key no. 2 `40BC86BC`? or `00000015`?
thanks for any future clarification!

**Comment by hozuki** [comment]
> @Lakoyo:
> in my case, it worked by drag and dropping HCACC.exe to the CMD window. I made sure to leave a space after ".exe" though. And then drag and drop your HCA file, this should specify exactly to where the input HCA file is. And if you want to save it next to your input HCA, then copy-paste the path of your input HCA and change the HCA's file name. So it should look like this
> 
> `C:\Users\CURRENT USER>D:\More\DereTore-master\Apps\Hcacc\bin\Release\hcacc.exe "D:\more\custom_song\input hca.hca" "D:\more\custom_song\output hca.hca" -ot 0` and just hit enter.
> 
> And actually, I want to point out that @hozuki 's answer on #51 raised another question for me. Which 16-hex halves go to where? I got a bit confused when they considered the second half of the key to be 00000000. Now, my problem would be,
> what if the custom key's hex value was longer?
> I'll make my own custom key for this example:
> 91280410300 = 1540BC86BC
> so is key no. 1 `00000015`? or `40BC86BC`?
> or is key no. 2 `40BC86BC`? or `00000015`?
> thanks for any future clarification!

Key 1 is the lower 32 bits (`40bc86bc`) and key 2 is the higher 32 bits (`00000015`).

**Comment by SonicSpace** [comment]
So I'm trying to create a mod for a game called Sonic Colors Ultimate and I need to encrypt the HCA files with the following key: 1991062320101111

Do you know how I could do this? Thanks!
I've tried in every possible way using your program, but without success the audio is all distorted in the game
